homeassistant/components/mjpeg/config_flow.py

```diff
@@ -0,0 +1,258 @@
+
+        if user_input is not None:
+            errors, authentication = await async_validate_input(self.hass, user_input)
+            if not errors:
+                for entry in self._async_current_entries(include_ignore=False):
+                    if entry.options[CONF_MJPEG_URL] == user_input[CONF_MJPEG_URL]:
+                        return self.async_abort(reason="already_configured")
+
+                # Storing data in option, to allow for changing them later
+                # using an options flow.
+                return self.async_create_entry(
+                    title=user_input.get(CONF_NAME, user_input[CONF_MJPEG_URL]),
+                    data={},
+                    options={
+                        CONF_AUTHENTICATION: authentication,
+                        CONF_MJPEG_URL: user_input[CONF_MJPEG_URL],
+                        CONF_PASSWORD: user_input[CONF_PASSWORD],
+                        CONF_STILL_IMAGE_URL: user_input.get(CONF_STILL_IMAGE_URL),
+                        CONF_USERNAME: user_input.get(CONF_USERNAME),
+                        CONF_VERIFY_SSL: user_input[CONF_VERIFY_SSL],
+                    },
+                )
+        else:
+            user_input = {}
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=async_get_schema(user_input, show_name=self.show_name),
+            errors=errors,
+        )
+
+    async def async_step_import(self, config: dict[str, Any]) -> FlowResult:
+        """Handle a flow initialized by importing a config."""
+        for entry in self._async_current_entries(include_ignore=False):
+            if entry.options[CONF_MJPEG_URL] == config[CONF_MJPEG_URL]:
+                return self.async_abort(reason="already_configured")
+
+        return self.async_create_entry(
+            title=config[CONF_NAME],
+            data={},
+            options={
+                CONF_AUTHENTICATION: config[CONF_AUTHENTICATION],
+                CONF_MJPEG_URL: config[CONF_MJPEG_URL],
+                CONF_PASSWORD: config[CONF_PASSWORD],
+                CONF_STILL_IMAGE_URL: config[CONF_STILL_IMAGE_URL],
+                CONF_USERNAME: config.get(CONF_USERNAME),
+                CONF_VERIFY_SSL: config[CONF_VERIFY_SSL],
+            },
+        )
+
+
+class MJPEGOptionsFlowHandler(OptionsFlow):
+    """Handle MJPEG IP Camera options."""
+
+    show_name: bool = False
+
+    def __init__(self, config_entry: ConfigEntry) -> None:
+        """Initialize MJPEG IP Camera options flow."""
+        self.config_entry = config_entry
+
+    async def async_step_init(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Manage MJPEG IP Camera options."""
+        errors: dict[str, str] = {}
+
+        if user_input is not None:
+            errors, authentication = await async_validate_input(self.hass, user_input)
+            if not errors:
+                for entry in self.hass.config_entries.async_entries(DOMAIN):
```

Comment:
I was going to suggest `_async_abort_entries_match` but it only works for `.data`

I think it would be safe to match against options as well though

Reply:
Hehe I totally understand this comment. I wrote it using `_async_abort_entries_match`, and wondered why it wasn't working 🙈

Reply:
https://github.com/home-assistant/core/pull/66662